src/libbson/CMakeLists.txt

```diff
@@ -236,8 +224,25 @@ set (HEADERS_FORWARDING
 
 add_library (bson_shared SHARED ${SOURCES} ${HEADERS} ${HEADERS_FORWARDING})
 set (CMAKE_CXX_VISIBILITY_PRESET hidden)
-target_compile_definitions (bson_shared PRIVATE BSON_COMPILATION JSONSL_PARSE_NAN)
-target_include_directories (bson_shared INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src> $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/src> $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/src/bson>)
+target_compile_definitions (bson_shared
+   PRIVATE
+      BSON_COMPILATION
+      JSONSL_PARSE_NAN
+      COMMON_PREFIX_=_bson_common
+)
+# Several directories in the source and build trees contain headers we would like
+# include via relative reference, but they all end up in the same install path
+target_include_directories (
+   bson_shared
+   PUBLIC
```

Comment:
Previously, some directories had `INTERFACE`, and the ones included through `include_directories` are implicitly 
 private. But AFAICT, the `PUBLIC` scope does not have any observable difference. You had mentioned in slack that cmake won't add extraneous directories in the installed package scripts. And I compared the install directories before and after these changes to confirm.